umake/frameworks/__init__.py

```diff
@@ -340,11 +341,57 @@ def load_module(module_abs_name, main_category):
     if current_category not in BaseCategory.categories.values():
         return
     for framework_name, FrameworkClass in inspect.getmembers(module, _is_frameworkclass):
-        if FrameworkClass(current_category) is not None:
+        if FrameworkClass(category=current_category, force_loading=force_loading) is not None:
             logger.debug("Attach framework {} to {}".format(framework_name, current_category.name))
 
 
-def load_frameworks():
+def list_frameworks():
+    """List all frameworks.
+
+        Return a json dictionnary. Format of the json string is:
+        { 'category_name':
+          [
+            { 'framework_name':
+                {
+                   'detail': detail_name (always filled, then, the output will filter it if detail isn't chosen)
+                   'install_path': None or path string
+                   'installable': True or False
+                   'only_for_removal': True or False
+                }
+            },
+          ]
+        }"""
+    cat_dict = list()
+    for category in BaseCategory.categories.values():
+        new_cat = dict()
+
+        new_cat["name"] = category.prog_name
+        new_cat["description"] = category.description
+
+        framework_dict = list()
+        for framework in category.frameworks.values():
+            new_fram = dict()
+
+            new_fram["name"] = framework.prog_name
```

Comment:
Another (less repetitive form) would be:
```
new_fram = {
  "name": framework.prog_name,
  "only_for_removal": framework.only_for_removal,
  …
}
```
same for category creation above (and that way, you even don't need `= dict()` ;)